### PR TITLE
Implement HallController, ShoppingCartController, OrderController, Us…

### PR DESCRIPTION
**Pull request by KovalAlexey**
…erController

### CODE REVIEW

**Review by vladkuzya — APPROVED** [review]
Good!)

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public ShoppingCartController(ShoppingCartMapper mapper, ShoppingCartService cartService,
+                                  UserService userService, MovieSessionService sessionService) {
+        this.mapper = mapper;
+        this.cartService = cartService;
+        this.userService = userService;
+        this.sessionService = sessionService;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getUserById(@RequestParam Long id) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]:
```suggestion
    public ShoppingCartResponseDto getByUserId(@RequestParam Long id) {
```

File: src/main/java/com/dev/cinema/dao/TicketDao.java [hunk 2]
```diff
@@ -4,4 +4,6 @@
 
 public interface TicketDao {
     Ticket add(Ticket ticket);
+
+    Ticket getById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/TicketDao.java]: where do you use it?

File: src/main/java/com/dev/cinema/service/mapper/ShoppingCartMapper.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.dto.ShoppingCartResponseDto;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartMapper {
+    public ShoppingCartResponseDto convertToResponseDto(ShoppingCart order) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/ShoppingCartMapper.java]: why order?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/ShoppingCartController.java (hunk 1 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/ShoppingCartController.java]: done

File: src/main/java/com/dev/cinema/service/mapper/ShoppingCartMapper.java (hunk 3 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/service/mapper/ShoppingCartMapper.java]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**